doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -611,9 +614,9 @@ The following syntactic sugar is provided for two common cases:
 Strict vs. non-strict mode
 ++++++++++++++++++++++++++
 
-Depending on the context, quotation-producing terms (i.e. ``constr`` or
-``open_constr``) are not internalized in the same way. There are two possible
-modes, the *strict* and the *non-strict* mode.
+Depending on the context, quotation-producing terms (i.e. ``constr``,
+``open_constr`` or ``preterm``) are not internalized in the same way.
+There are two possible modes, the *strict* and the *non-strict* mode.
```

Comment:
```suggestion
Terms can be internalized in two ways: *strict* and *non-strict*, depending
on the context.
```

Then explain how  ``constr``, ``open_constr`` and ``preterm`` are handled after the explanation of the modes/ways.

I'd also change "mode" to "way" in this subsection because it's not a settable mode, is it?

Reply:
I don't see how that would be better.

Reply:
"not internalized in the same way" is an awkward way of saying that they are internalized differently.  Since you mention there is a difference, it should be described briefly somewhere.  How are users supposed to understand which constructs are strict or non-strict?  How do know what strict and non-strict mean?

Reply:
todo for some other PR, make an issue if you want to track it

Reply:
Track it, as in you'll do it later?  If you know the technical details, you should spend the few minutes needed now to briefly document them.  Poorly described functionality probably won't be used.  Everyone needs to take some responsibility for the documentation.  I certainly can't do all of it and I usually need help from others to get details right.

Reply:
As in I'm not doing it in this PR and I'm not commiting to doing it in the future.

Reply:
I think you are talking about different things.  @jfehrle, "``constr``,
``open_constr`` or ``preterm``" is a definition of what it means for an Ltac2 term to be "quotation-producing".  If you write `true` or `()` or `f x` in Ltac2, these are not quotation-producing.  The Ltac2 terms that are quotation-producing are the terms that are interpreted in Ltac2-scope constr or open_constr.  This PR additionally adds preterm scope, which is also quotation producing.  This section is not about the differences between constr, open_constr, and preterm; this section is about the difference between strict and non-strict mode.  The difference is [already described in the existing documentation](https://coq.inria.fr/refman/proof-engine/ltac2.html#strict-vs-non-strict-mode).  This PR is not changing anything of significance here, just adding "preterm" to the list of quotation-producing internalization scopes, and so this PR is not the proper place to make wording suggestions about explaining strict mode.

The differences between constr, open_constr, and preterm are described above:
```
- ``open_constr``, which parses Coq terms and produces a term potentially with
  holes at runtime (type ``Init.constr`` as well).
- ``preterm``, which parses Coq terms and produces a value which must
  be typechecked with ``Constr.pretype`` (type ``Init.preterm``).
```
Here "parsing a term and producing a term/value" is a roughly synonymous with "internalization"

Reply:
Ah, good point. Then it would make sense to hyperlink to the "Strict vs. non-strict mode" section.  Either as a `:ref:` or add the two terms to the glossary (`:gdef:`/`:term:`).

Reply:
@jfehrle This change is itself of the first paragraph of the strict vs non-strict section.

Reply:
OK, my bad, nevermind.